examples/read_bids_datasets.py

```diff
@@ -0,0 +1,183 @@
+        edf_path = eegbci.load_data(subject=subj_idx, runs=task_idx)[0]
+
+        raw = mne.io.read_raw_edf(edf_path, preload=False, stim_channel=None)
+        annot = mne.read_annotations(edf_path)
+        raw.set_annotations(annot)
+        events, event_id = mne.events_from_annotations(raw)
+
+        bids_basename = make_bids_basename(subject='{:03}'.format(subj_idx),
+                                           task=task_names[task_idx],
+                                           run=run_mapping[task_idx])
+
+        write_raw_bids(raw, bids_basename, bids_root, event_id=trial_type,
+                       events_data=events, overwrite=True)
+
+###############################################################################
+# Print the directory tree
+print_dir_tree(bids_root)
+
+###############################################################################
+# Step 2: Read a BIDS dataset
+# ---------------------------
+#
+# Let's read in the dataset and show off a few features of the
+# loading function `read_raw_bids`.
+bids_fname = bids_basename + "_eeg.edf"
+raw = read_raw_bids(bids_fname, bids_root, verbose=True)
+
+###############################################################################
+# `raw.info` has the basic subject metadata
+print(raw.info['subject_info'])
+
+# `raw.info` has the basic channel metadata
+print(raw.info['dig'])
+
+# `raw.info` has the PowerLineFrequency loaded in
+print(raw.info['line_freq'])
+
+# annotations
+print(raw.annotations)
+
+# channel info dictionary
+print(raw.info['chs'])
+
+###############################################################################
+# Let's modify data in some of the sidecar files and re-load the data
+# and see how the `raw.info` has changed.
+###############################################################################
+# modify sidecar json
+sidecar_fname = _find_matching_sidecar(bids_fname, bids_root, 'eeg.json')
+with open(sidecar_fname, 'r') as fin:
+    sidecar_json = json.load(fin)
+sidecar_json["PowerLineFrequency"] = 75  # for the sake of example
+with open(sidecar_fname, 'w') as fout:
+    json.dump(sidecar_json, fout)
+
+# check new raw read in
+raw = read_raw_bids(bids_fname, bids_root, verbose=True)
+print(raw.info['line_freq'])
+
+# modify channels tsv
+channels_fname = _find_matching_sidecar(bids_fname, bids_root, 'channels.tsv')
+channels_dict = _from_tsv(channels_fname)
+channels_dict['type'][0] = 'MISC'
+channels_dict['type'][0] = 'EEG'
+_to_tsv(channels_dict, channels_fname)
+
+# check new raw read in
+raw = read_raw_bids(bids_fname, bids_root, verbose=True)
+print(raw.info['chs'])
+
+# modify events tsv
+events_fname = _find_matching_sidecar(bids_fname, bids_root, 'events.tsv')
+# Create an arbitrary events
+events = {'onset': [11, 12, 13],
+          'duration': ['n/a', 'n/a', 'n/a']
+          }
+_to_tsv(events, events_fname)
+
+# check new raw read in
+raw = read_raw_bids(bids_fname, bids_root, verbose=True)
+print(raw.annotations)
```

Comment:
This example is way too long as it stands. The somato data in MNE is already BIDS compatible. Should we use that instead so that the example focuses only on reading and is not more than 10-15 lines long? We should not intimidate new users with code.